docs/standalone.md

```diff
@@ -0,0 +1,354 @@
+- `VALKEY_REPLY_PUSH` - An out of band push reply. This is also array-like in nature.
+
+### Disconnecting / cleanup
+
+When libvalkey returns non-null `valkeyRepy` struts you are responsible for freeing them with `freeReplyObject`.  In order to disconnect and free the context simply call `valkeyFree`.
+
+```c
+valkeyReply *reply = valkeyCommand(ctx, "set %s %s", "foo", "bar");
+// Error handling ...
+freeReplyObject(reply);
+
+// Disconnect and free context
+valkeyFree(ctx);
+```
+
+### Pipelining
+
+`valkeyCommand` and `valkeyCommandArgv` each make a round-trip to the server, by sending the command and then waiting for a reply`. Alternatively commands may be pipelined with the `valkeyAppendCommand` and `valkeyAppendCommandArgv` functions.
+
+When you use `valkeyAppendCommand` the command is simply appended to the output buffer of `valkeyContext` but not delivered to the server, until you attempt to read the first response, at which point the entire buffer will be delivered.
+
+```c
+// No data will be delivered to the server while these commands are being appended.
+for (size_t i = 0; i < 100000; i++) {
+    if (valkeyAppendCommand(c, "INCRBY key:%zu %zu", i, i) != VALKEY_OK) {
+        fprintf(stderr, "Error appending command: %s\n", c->errstr);
+        exit(1);
+    }
+}
+
+// The entire output buffer will be delivered on the first call to `valkeyGetReply`.
+for (size_t i = 0; i < 100000; i++) {
+    if (valkeyGetReply(c, (void**)&reply) != VALKEY_OK) {
+        fprintf(stderr, "Error reading reply %zu: %s\n", i, c->errstr);
+        exit(1);
+    } else if (reply->type != VALKEY_REPLY_INTEGER) {
+        fprintf(stderr, "Error:  Non-integer reply to INCRBY?\n");
+        exit(1);
+    }
+
+    printf("INCRBY key:%zu => %lld\n", i, reply->integer);
+    freeReplyObject(reply);
+}
+```
+
+`valkeyGetReply` can also be used in other contexts than pipeline, for example when you want to continuously block for commands for example in a subscribe context.
+
+```c
+valkeyReply *reply = valkeyCommand(c, "SUBSCRIBE channel");
+assert(reply != NULL && !c->err);
+
+while (valkeyGetReply(c, (void**)&reply) == VALKEY_OK) {
+    // Do something with the message...
+    freeReplyObject(reply);
+}
+```
+
+### Errors
+
+As previously mentioned, when there is a communication error libvalkey will return `NULL` and set the `err` and `errstr` members with the nature of the problem. The specific error types are as follows.
+
+- `VALKEY_ERR_IO` - A problem with the connection.
+- `VALKEY_ERR_EOF` - The server closed the connection.
+- `VALKEY_ERR_PROTOCOL` - There was an error parsing the reply.
+- `VALKEY_ERR_TIMEOUT` - A connect, read, or write timeout.
+- `VALKEY_ERR_OOM` - Out of memory.
+- `VALKEY_ERR_OTHER` - Some other error (check `c->errstr` for details).
+
+### Thread safety
+
+Libvalkey context structs are not thread safe. You should not attempt to share them between threads, unless you really know what you're doing.
```

Comment:
```suggestion
Libvalkey context structs are **not** thread safe. You should not attempt to share them between threads, unless you really know what you're doing.
```
since there has been a lot of thread related issues  :)